extern/ZAPDUtils/Utils/StringHelper.cpp

```diff
@@ -140,6 +142,81 @@ bool StringHelper::HasOnlyDigits(const std::string& str)
 	return std::all_of(str.begin(), str.end(), ::isdigit);
 }
 
+bool HasOnlyHexDigits(const std::string& str)
+{
+	if (str.length() == 0)
+    {
+        return false;
+    }
+    else if(str.length() == 1)
+    {
+        return ::isdigit(str[0]);
+    }
+    else if (str.length() == 2)
+    {
+        if (str[1] == 'x' || str[1] == 'X')
+        {
+            return false;
+        }
+        else
+        {
+            return std::all_of(str.begin(), str.end(), ::isdigit);
+        }
+    } 
+    else if (str.length() >= 3)
+    {
+        if (str[1] == 'x' || str[1] == 'X')
+        {
+            return std::all_of(str.begin() + 2, str.end(), ::isxdigit);
+        }
+        else
+        {
+            return false;
+        }
+    }
+    else
+    {
+        return false; // ??
+    }
+}
+bool HasOnlyHexDigits(std::string_view str) 
+{
+	if (str.length() == 0)
+    {
+        return false;
+    }
+    else if(str.length() == 1)
+    {
+        return ::isdigit(str[0]);
+    }
+    else if (str.length() == 2)
+    {
+        if (str[1] == 'x' || str[1] == 'X')
+        {
+            return false;
+        }
+        else
+        {
+            return std::all_of(str.begin(), str.end(), ::isdigit);
+        }
+    } 
+    else if (str.length() >= 3)
+    {
+        if (str[1] == 'x' || str[1] == 'X')
+        {
+            return std::all_of(str.begin() + 2, str.end(), ::isxdigit);
+        }
+        else
+        {
+            return false;
+        }
+    }
+    else
+    {
+        return false; // ??
+    }
+}
+
```

Comment:
feels like a lot of code when there are simpler solutions
https://stackoverflow.com/questions/8899069/how-to-find-if-a-given-string-conforms-to-hex-notation-eg-0x34ff-without-regex

i'm a fan of the "call [strtoul](http://linux.die.net/man/3/strtoul) and check for an error" method

Reply:
if it's not possible to simplify this, then adding comments explaining why it's complex would be good. also, it seems like both methods are identical except one is taking in a view and the other is taking in a string, it'd be nice to not have copypasta